src/main/java/seedu/address/model/util/predicate/CompositeFieldPredicate.java

```diff
@@ -37,9 +38,16 @@ public boolean test(U u) {
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-            || (other instanceof CompositeFieldPredicate // instanceof handles nulls
-            && fieldPredicateSet.equals(((CompositeFieldPredicate<?>) other).fieldPredicateSet)); // state check
```

Comment:
Is the original set equals not working?  I think the original equals for set do checks the individual elements according to the API.